### PR TITLE
Organize helper functions / utilities

### PR DESCRIPTION
**Pull request by dylanhmorris**
- Move some reused R code into `hewr` and depend on it in pipeline scripts
- Clean up some python pipelining code and restore a needed helper function for parsing directories.

### CODE REVIEW

File: hewr/R/directory_utils.R [hunk 1]
```diff
@@ -61,3 +68,39 @@ parse_model_run_dir <- function(model_run_dir_path) {
     parse_model_batch_dir(batch_dir)
```
Comment (damonbayer) [inline hewr/R/directory_utils.R:68]:
```suggestion
    parse_model_batch_dir_name(batch_dir)
```

Seems like this should have been caught by a test.

**Comment by damonbayer** [comment]
Flagging that this will impact  https://github.com/CDCgov/pyrenew-hew/pull/136

File: hewr/R/directory_utils.R [hunk 2]
```diff
@@ -61,3 +68,39 @@ parse_model_run_dir <- function(model_run_dir_path) {
...
   ))
 }
+
+
+#' Get forecast directories.
```
Comment (damonbayer) [inline hewr/R/directory_utils.R:73]: I think we had been calling these `model_batch` directories?

File: pipelines/create_hubverse_table.R [hunk 3]
```diff
@@ -66,9 +66,9 @@ draws_to_quantiles <- function(forecast_dir,
   return(epiweekly_quantiles)
 }
 
-create_hubverse_table <- function(model_run_dir,
+create_hubverse_table <- function(model_batch_dir,
```
Comment (damonbayer) [inline pipelines/create_hubverse_table.R:69]: should this just go in `hewr`?

File: hewr/R/directory_utils.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (dylanhmorris) [reply hewr/R/directory_utils.R]: Fixed in 83a1d65

Comment (dylanhmorris) [reply hewr/R/directory_utils.R]: Fixed and test added in 83a1d65

File: hewr/R/directory_utils.R (hunk 2 above)
Comment (dylanhmorris) [reply hewr/R/directory_utils.R:73]: Made consistent in 83a1d65

File: pipelines/create_hubverse_table.R (hunk 3 above)
Comment (dylanhmorris) [reply pipelines/create_hubverse_table.R:69]: Probably, yes.

Comment (dylanhmorris) [reply pipelines/create_hubverse_table.R]: Done in 667b49b

File: hewr/R/directory_utils.R (hunk 1 above)
Comment (dylanhmorris) [reply hewr/R/directory_utils.R]: Fixed and tests added as of 3949062

File: hewr/R/directory_utils.R [hunk 4]
```diff
@@ -37,12 +36,29 @@ parse_model_batch_dir_name <- function(model_batch_dir_name) {
...
-    report_date = lubridate::ymd(matches[3]),
-    first_training_date = lubridate::ymd(matches[4]),
-    last_training_date = lubridate::ymd(matches[5])
-  ))
+    report_date = lubridate::ymd(matches[3], quiet = TRUE),
+    first_training_date = lubridate::ymd(matches[4], quiet = TRUE),
+    last_training_date = lubridate::ymd(matches[5], quiet = TRUE)
+  )
+
+  if (any(sapply(result, is.null)) || any(sapply(result, is.na))) {
```
Comment (damonbayer) [inline hewr/R/directory_utils.R:46]:
```suggestion
  if (any(sapply(result, is.null)) || any(is.na(result))) {
```
`is.na` works as expected with lists.

Comment (damonbayer) [reply hewr/R/directory_utils.R:46]: You would need 
```r
disease_map_lower <- c(
  "covid-19" = "COVID-19",
  "influenza" = "Influenza"
)
```
and
```r
parse_model_batch_dir_path <- function(model_batch_dir_path) {
  pattern <- "(.+)_r_(.+)_f_(.+)_t_(.+)"
  model_batch_dir_name <- fs::path_file(model_batch_dir_path)
  matches <- stringr::str_match(
    model_batch_dir_name,
    pattern
  )
  
  if (any(is.na(matches))) {
    stop(
      "Invalid format for model batch directory name; ",
      "could not parse. Expected ",
      "'<disease>_r_<report_date>_f_<first_training_date>_t_",
      "<last_training_date>'."
    )
  }
  
  result <- list(
    disease = disease_map_lower[matches[2]],
    report_date = lubridate::ymd(matches[3], quiet = TRUE),
    first_training_date = lubridate::ymd(matches[4], quiet = TRUE),
    last_training_date = lubridate::ymd(matches[5], quiet = TRUE)
  )
  
  if (any(is.na(result))) {
    stop(
      "Could not parse extracted disease and/or date ",
      "values expected 'disease' to be one of 'covid-19' ",
      "and 'influenza' and all dates to be valid dates in ",
      "YYYY-MM-DD format. Got: ",
      glue::glue(
        "disease: {matches[[2]]}, ",
        "report_date: {matches[[3]]}, ",
        "first_training_date: {matches[[4]]}, ",
        "last_training_date: {matches[[5]]}."
      )
    )
  }
  return(result)
}
```

**Review by damonbayer — APPROVED** [review]
Thanks @dylanhmorris! One opportunity to simplify things.

**Pull request merged**